doc/reference/reference_lua/http.rst

```diff
@@ -1,353 +1,1141 @@
+:ref:`get <client_object-get>`, :ref:`post <client_object-post>`, :ref:`put <client_object-put>`, and so on.
+For example, you can replace the request above by calling ``get`` as follows:
+
+..  literalinclude:: /code_snippets/test/http_client/get_test.lua
+    :language: lua
+    :lines: 1-2
+
+
+
+.. _request_query_parameters:
+
+Query parameters
+~~~~~~~~~~~~~~~~
+
+To add query string parameters, use the :ref:`params <request_options-params>` option exposed by the :ref:`request_options <request_options>` object:
+
+..  literalinclude:: /code_snippets/test/http_client/get_add_query_params_test.lua
+    :language: lua
+    :lines: 1-5
+
+In the example above, the requested URL is ``https://httpbin.org/get?page=1``.
+
+.. NOTE::
+
+    If a parameter name or value contains a reserved character (for example, ``&`` or ``=``),
+    the HTTP client `encodes <https://en.wikipedia.org/wiki/URL_encoding>`_ a query string.
+
+
+.. _request_headers:
+
+Headers
+~~~~~~~
+
+To add headers to the request, use the :ref:`headers <request_options-headers>` option:
+
+..  literalinclude:: /code_snippets/test/http_client/get_add_header_test.lua
+    :language: lua
+    :lines: 1-8
+
+
+.. _request_cookies:
+
+Cookies
+~~~~~~~
+
+You can add cookies to the request using the :ref:`headers <request_options-headers>` option:
+
+..  literalinclude:: /code_snippets/test/http_client/get_add_cookie_test.lua
+    :language: lua
+    :lines: 1-7
+
+To learn how to obtain cookies passed in the ``Set-Cookie`` response header, see :ref:`Response cookies <response_cookies>`.
+
+
+
+.. _request_body:
+
+Body
+~~~~
+
+.. _request_serialization:
+
+Serialization
+*************
+
+The HTTP client automatically serializes the content in a specific format when sending a request based on the specified ``Content-Type`` header.
+By default, the client uses the ``application/json`` content type and sends data serialized as JSON:
+
+..  literalinclude:: /code_snippets/test/http_client/post_json_test.lua
+    :language: lua
+    :lines: 1-6
+
+The body for the request above might look like this:
+
+.. code-block:: console
+
+    {
+        "user_id": 123,
+        "user_name": "John Smith"
+    }
+
+To send data in a YAML or MsgPack format, set the ``Content-Type`` header explicitly to ``application/yaml`` or ``application/msgpack``, for example:
+
+..  literalinclude:: /code_snippets/test/http_client/post_yaml_test.lua
+    :language: lua
+    :lines: 1-10
+
+In this case, the request body is serialized to YAML:
+
+.. code-block:: console
+
+    user_id: 123
+    user_name: John Smith
+
+
+.. _request_form_parameters:
+
+Form parameters
+***************
+
+To send form parameters using the ``application/x-www-form-urlencoded`` type,
+use the :ref:`params <request_options-params>` option:
+
+..  literalinclude:: /code_snippets/test/http_client/post_form_params_test.lua
+    :language: lua
+    :lines: 1-5
+
+
+.. _request_streaming_upload:
+
+Streaming upload
+****************
```

Comment:
I would like to highlight here that the client can optionally set the `Content-Length` header manually:

https://github.com/tarantool/tarantool/blob/53c94bc79ecfc0451bc3ffa57ffb88e3217671ff/test/app-luatest/http_client_test.lua#L1225

In that case a regular HTTP POST method will be used instead of  https://en.wikipedia.org/wiki/Chunked_transfer_encoding .

Reply:
thanks, will do